src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -3826,18 +3830,32 @@ class C
     }
 }", finalText)
 
-                ' This should have happened as two text changes to the buffer.
                 Dim changes = snapshotBeforeCommit.Version.Changes
-                Assert.Equal(2, changes.Count)
-
-                Dim actualChanges = changes.ToArray()
-                Dim firstChange = actualChanges(0)
-                Assert.Equal(New Span(0, 0), firstChange.OldSpan)
-                Assert.Equal("using NewUsing;", firstChange.NewText)
-
-                Dim secondChange = actualChanges(1)
-                Assert.Equal(New Span(testDocument.CursorPosition.Value, 0), secondChange.OldSpan)
-                Assert.Equal("InsertedItem", secondChange.NewText)
+                Select Case completionImplementation
+                    Case CompletionImplementation.Legacy ' Specific of the legacy implementation
+                        ' This should have happened as two text changes to the buffer.
+                        Assert.Equal(2, changes.Count)
+
+                        Dim actualChanges = changes.ToArray()
+                        Dim firstChange = actualChanges(0)
+                        Assert.Equal(New Span(0, 0), firstChange.OldSpan)
+                        Assert.Equal("using NewUsing;", firstChange.NewText)
+
+                        Dim secondChange = actualChanges(1)
+                        Assert.Equal(New Span(testDocument.CursorPosition.Value, 0), secondChange.OldSpan)
+                        Assert.Equal("InsertedItem", secondChange.NewText)
+                    Case CompletionImplementation.Modern
+                        Assert.Equal(1, changes.Count)
+                        Dim actualChanges = changes.ToArray()
+                        Dim firstChange = actualChanges(0)
+                        Assert.Equal(New Span(0, testDocument.CursorPosition.Value), firstChange.OldSpan)
+                        Assert.Equal("using NewUsing;
+using System;
+class C
+{
+    void goo() {
+        return InsertedItem", firstChange.NewText)
```

Comment:
this issue still seems like it needs to be resolved.

Reply:
Note: this test is literally called "TestLargeChangeBrokenUpIntoSmallTextChanges".  It exists to validate this behavior. But hte new behavior is incorrect.

Reply:
Yes, this is still under investigations: we will review it once again before enabling the feature flag. Here is an issue to track: [https://github.com/dotnet/roslyn/issues/30972](https://github.com/dotnet/roslyn/issues/30972)

---
In reply to: [232508808](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508808) [](ancestors = 232508808)

Reply:
We'll come back to review the issue with [https://github.com/dotnet/roslyn/issues/30972](https://github.com/dotnet/roslyn/issues/30972)

---
In reply to: [232508751](https://github.com/dotnet/roslyn/pull/29016#discussion_r232508751) [](ancestors = 232508751)